manager/controllers/app/blueprint_mgr.go

```diff
@@ -24,49 +19,6 @@ func containsTemplate(templateList []app.ComponentTemplate, moduleName string) b
 	return false
 }
 
-// GetBlueprintSignature returns the signature (name and namespace) of the blueprint owned by the given M4DApplication instance
-func (r *M4DApplicationReconciler) GetBlueprintSignature(applicationContext *app.M4DApplication) *app.Blueprint {
-	return &app.Blueprint{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      applicationContext.Name,
-			Namespace: applicationContext.Status.BlueprintNamespace,
-		},
-	}
-}
-
-// getBlueprint finds the Blueprint instance associated with this M4DApplication if one already exists,
-// and creturns it
-func (r *M4DApplicationReconciler) getBlueprint(applicationContext *app.M4DApplication) (*app.Blueprint, error) {
-	// Check if there is an allocated namespace for the blueprint
-	if applicationContext.Status.BlueprintNamespace == "" {
-		return nil, nil
-	}
-	// Get the Blueprints owned by this M4DApplication instance
-	m4dBlueprint := r.GetBlueprintSignature(applicationContext)
-	namespace, _ := client.ObjectKeyFromObject(m4dBlueprint)
-	if err := r.Client.Get(context.Background(), namespace, m4dBlueprint); err != nil {
-		return nil, err
-	}
-	return m4dBlueprint, nil
-}
-
-// DeleteOwnedBlueprint deletes the blueprint owned by the given M4DApplication instance,
-// checking first that it actually exists
-func (r *M4DApplicationReconciler) DeleteOwnedBlueprint(applicationContext *app.M4DApplication) error {
-	// Check first if the blueprint exists
-	blueprint, _ := r.getBlueprint(applicationContext)
-	if blueprint == nil {
-		r.Log.V(0).Info("\tDidn't find a blueprint to delete: " + applicationContext.GetName())
-		return nil
-	}
-
-	if err := r.Delete(context.Background(), blueprint); err != nil {
-		r.Log.V(0).Info("\tFailed to delete blueprint: " + applicationContext.GetName())
-		return err
-	}
-	return nil
-}
-
```

Comment:
While you have generalized these functions, GenerateBlueprint is still tightly coupled with M4DApplicationReconciler.  This is fine for now but as per our discussion regarding future flows we may want to create a blueprint / plotter from a controller other than M4DApplication. This is future though, no changes needed now.

Reply:
Will be rewritten. M4DApplication object is used for copying a selector (do we need this in a Plotter/Blueprint?) and reusing the m4dapplication name which can be done using a regular runtime.Object.